src/transport_ofi.c

```diff
@@ -315,9 +358,46 @@ int bind_enable_cntr_ep_resources(shmem_transport_ctx_t *ctx)
 {
     int ret = 0;
 
-    /* Attach the shared context */
-    ret = fi_ep_bind(ctx->cntr_ep, &ctx->stx->fid, 0);
-    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind STX to CNTR endpoint failed");
+    /* STX contexts are shared by SHMEM contexts that are private to the same
+     * thread (i.e. have SHMEM_CTX_PRIVATE option set and same thread ID).  */
+    if (ctx->options & SHMEM_CTX_PRIVATE) {
+        shmem_transport_ofi_stx_kvs_t *e = malloc(sizeof(shmem_transport_ofi_stx_kvs_t));
+        e->tid = ctx->tid;
+        e->stxs = malloc(sizeof(shmem_transport_ofi_stx_t));
```

Comment:
I think you want e->stxs to be a pointer rather than a struct, so you can capture a reference to the STX rather than a deep copy of the struct (you aren't going to succeed at deep copying the OFI stx object, so better not to try).

Reply:
+1